eth/fetcher/tx_fetcher.go

```diff
@@ -355,17 +424,48 @@ func (f *TxFetcher) loop() {
 			f.scheduleFetches(timeoutTimer, nil)
 
 		case delivery := <-f.cleanup:
+			// If the request failed, reschedule all the hashes
+			if delivery.direct && delivery.failed {
```

Comment:
I think `direct` is unintuitive. Is it delivered 'directly' via standard broadcast, or is it a 'direct' response to our query? Could you rename it to e.g `requested` or `broadcast` instead?